src/geom/elem.C

```diff
@@ -358,10 +360,13 @@ Real Elem::hmin() const
 {
   Real h_min=std::numeric_limits<Real>::max();
 
-  for (unsigned int n_outer=0; n_outer<this->n_vertices(); n_outer++)
-    for (unsigned int n_inner=n_outer+1; n_inner<this->n_vertices(); n_inner++)
+  // Avoid calling a virtual a lot of times
+  const auto n_vertices = this->n_vertices();
+
+  for (unsigned int n_outer=0; n_outer<n_vertices; n_outer++)
+    for (unsigned int n_inner=n_outer+1; n_inner<n_vertices; n_inner++)
       {
-        const Point diff = (this->point(n_outer) - this->point(n_inner));
+        const auto diff = (this->point(n_outer) - this->point(n_inner));
```

Comment:
Why bother changing this?

Reply:
Avoid an extra stack frame of constructor call? :laughing:

Reply:
Or maybe one day the `point` method may not return a `Point` ;-) I can definitely remove this change...

Reply:
Wait, does this actually remove a call?  diff will still get declared as a const Point, won't it?

Reply:
I just like avoiding auto except where it adds flexibility (e.g. in templated code) or removes verbosity (e.g. in iterators to crazy long nested container types) because it can remove type-safety (when otherwise the compiler might scream that the type you expected and the type you got aren't the same).

Reply:
> Wait, does this actually remove a call? diff will still get declared as a const Point, won't it?

No I believe it will actually be a `TypeVector` based on the `operator -` method that `Point` inherits from `TypeVector`

Reply:
I think the generated assembly for this line wouldn't change due to this diff. You might consider 
```
h_min = std::min(h_min, (this->point(n_outer) - this->point(n_inner)).norm_sq());
```
but again I wouldn't really expect a big difference.

Reply:
> I just like avoiding auto except where it adds flexibility (e.g. in templated code) or removes verbosity (e.g. in iterators to crazy long nested container types) because it can remove type-safety (when otherwise the compiler might scream that the type you expected and the type you got aren't the same).

That's fair. Explicit typing can also make the code more clear and legible to a later developer...